lint.go

```diff
@@ -991,46 +991,63 @@ func (f *file) lintVarDecls() {
 	})
 }
 
-// lintElses examines else blocks. It complains about any else block whose if block ends in a return.
+// lintElses examines else blocks. It complains about any else block whose if and else if blocks end in a return.
 func (f *file) lintElses() {
-	// We don't want to flag if { } else if { } else { } constructions.
+	// We don't want to flag if { } (else if { })+ else { } constructions.
 	// They will appear as an IfStmt whose Else field is also an IfStmt.
 	// Record such a node so we ignore it when we visit it.
-	ignore := make(map[*ast.IfStmt]bool)
+	ignore := make(map[*ast.IfStmt]struct{})
```

Comment:
revert. `map[T]bool` is the most common way to represent a set, and the space savings for using `map[T]struct{}` are smaller than people tend to assume.